packages/backend/src/modules/tracked-txs/modules/l2-costs/indexers/L2CostsPricesIndexer.ts

```diff
@@ -0,0 +1,81 @@
+import { CoingeckoId, UnixTime } from '@l2beat/shared-pure'
+
+import {
+  ManagedChildIndexer,
+  type ManagedChildIndexerOptions,
+} from '../../../../../tools/uif/ManagedChildIndexer'
+
+import { CoingeckoQueryService } from '@l2beat/shared'
+import {
+  L2CostsPricesRecord,
+  L2CostsPricesRepository,
+} from '../repositories/L2CostsPricesRepository'
+
+export interface L2CostsPricesIndexerDeps
+  extends Omit<ManagedChildIndexerOptions, 'name'> {
+  l2CostsPricesRepository: L2CostsPricesRepository
+  coingeckoQueryService: CoingeckoQueryService
+}
+
+export const ETHEREUM_COINGECKO_ID = CoingeckoId('ethereum')
+
+export class L2CostsPricesIndexer extends ManagedChildIndexer {
+  constructor(private readonly $: L2CostsPricesIndexerDeps) {
+    super({ ...$, name: 'l2_costs_prices' })
+  }
+
+  override async update(from: number, to: number): Promise<number> {
+    const unixFrom = new UnixTime(from)
+    const unixTo = new UnixTime(to)
+
+    const maxRange = unixFrom.add(
```

Comment:
The `CoingeckoQueryService` has a method `getAdjustedTo`. It's probably more convenient to use it here.

Reply:
Ahhh didn't know that. I was following PriceIndexer from TVL2